modules/renter/contractor/uptime_test.go

```diff
@@ -136,21 +144,26 @@ func TestIsOffline(t *testing.T) {
 		{[]modules.HostDBScan{oldBadScan, newGoodScan, currentBadScan}, false},
 		// data covers large range, no scans succeded
 		{[]modules.HostDBScan{oldBadScan, newBadScan, currentBadScan}, true},
-		// recent scan was good (and within uptimeWindow of oldBadScan, but that shouldn't matter)
-		{[]modules.HostDBScan{oldBadScan, oldBadScan, oldBadScan, oldGoodScan}, false},
-		// recent scan was good (and outside uptimeWindow of oldBadScan, but that shouldn't matter)
-		{[]modules.HostDBScan{oldBadScan, oldBadScan, oldBadScan, currentGoodScan}, false},
+		// old scan was good, recent scans are bad.
+		{[]modules.HostDBScan{oldGoodScan, newBadScan, newBadScan, currentBadScan}, true},
+		// recent scan was good, with many recent bad scans.
+		{[]modules.HostDBScan{oldBadScan, newGoodScan, newBadScan, currentBadScan, currentBadScan}, false},
+		// recent scan was good, old scans were bad.
+		{[]modules.HostDBScan{oldBadScan, newBadScan, currentBadScan, currentGoodScan}, false},
```

Comment:
Added some extra tests here because I felt the existing testing was incomplete. Implementation seems solid, but you should double check that the tests make sense.